src/api/routes/test.test.ts

```diff
@@ -23,28 +27,113 @@ describe('test endpoints', () => {
   });
 
   describe('GET /users/:id/tests', () => {
-    it('returns 401 if user is authenticated', async () => {
-      const id = new UserId();
-      await request(app).get(`/api/v1/users/${id.value}/tests`).expect(401);
+    it('returns 401 if user is not authenticated', async () => {
+      const userId = new UserId();
+      await request(app).get(`/api/v1/users/${userId.value}/tests`).expect(401);
     });
 
-    it('returns 200 with the existing test if user is found', async () => {
-      const user = await userRepository.save(
-        new User(new UserId(), new Email('harsh@example.com'))
-      );
+    it('returns 404 if user is not found', async () => {
+      const user = await userRepository.save(aNewUser());
+      const unknowUserId = new UserId();
 
-      const test_saved = await testRepository.save(
-        new Test(new TestId(), user.id, new TestTypeId(), new Date())
-      );
+      await request(app)
+        .get(`/api/v1/users/${unknowUserId.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .expect(404);
+    });
+
+    it('returns 200 with the existing test if user is found', async () => {
+      const user = await userRepository.save(aNewUser());
+      const test = await testRepository.save(aTest(user.id));
 
       await request(app)
         .get(`/api/v1/users/${user.id.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .expect(200)
+        .expect((response) => {
+          expect(response.body[0].id).toEqual(test.id.value);
+        });
+    });
+
+    it('returns 200 if a user with an access pass requests another users tests', async () => {
+      // TODO
+    });
+  });
+
+  describe('POST /users/:id/tests', () => {
+    it('returns 401 if user is not authenticated', async () => {
+      const userId = new UserId();
+      await request(app)
+        .post(`/api/v1/users/${userId.value}/tests`)
+        .expect(401);
+    });
+
+    it('returns 404 if user is not found', async () => {
+      const user = await userRepository.save(aNewUser());
+      const unknowUserId = new UserId();
+
+      await request(app)
+        .post(`/api/v1/users/${unknowUserId.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .expect(404);
+    });
+
+    it('returns 403 if a user with an access pass tries to create a test for another user', async () => {
```

Comment:
Should we not allow this? If I am a pharmasist that is entering a new test for someone shouldn't I be able to create a test for them?

Reply:
Yeah, typo.  It's fixed in the follow up PR that actually implements it.

Reply:
Should we fix it here so we can merge to master and move on.. Doesn't matter that much given it's coming right after. up to you

Reply:
Then I'll have to deal with a conflict so easier just to proceed!